instrumentation/liberty/liberty-dispatcher/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/liberty/dispatcher/LibertyDispatcherNetAttributesExtractor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.liberty.dispatcher;
+
+import io.opentelemetry.instrumentation.api.instrumenter.net.NetAttributesExtractor;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class LibertyDispatcherNetAttributesExtractor
+    extends NetAttributesExtractor<LibertyRequest, LibertyResponse> {
+
+  @Override
+  public String transport(LibertyRequest libertyRequest) {
+    return SemanticAttributes.NetTransportValues.IP_TCP;
+  }
+
+  @Override
+  public @Nullable String peerName(
+      LibertyRequest libertyRequest, @Nullable LibertyResponse libertyResponse) {
+    if (!libertyRequest.isCompleted()) {
```

Comment:
is checking `isCompleted()` a performance optimization? or is it needed for another reason?

Reply:
It is used to limit calling these methods to `onStart` because in `onEnd` they may throw a NPE.